src/dist/replication/lib/log_block.cpp

```diff
@@ -30,5 +30,45 @@ void log_block::append_mutation(const mutation_ptr &mu, const aio_task_ptr &cb)
     mu->write_to([this](const blob &bb) { add(bb); });
 }
 
+void log_appender::append_mutation(const mutation_ptr &mu, const aio_task_ptr &cb)
+{
+    if (_blk->size() > MAX_BLOCK_BYTES) {
+        _full_blocks_size += _blk->size();
+        _full_blocks_blob_cnt += _blk->data().size();
+        int64_t new_block_start_offset = _blk->start_offset() + _blk->size();
+        _full_blocks.push_back(std::move(_blk));
+        _blk = std::make_shared<log_block>(new_block_start_offset);
+    }
+    _blk->append_mutation(mu, cb);
+}
+
+std::shared_ptr<log_block> log_appender::take_log_block()
+{
+    if (!_full_blocks.empty()) {
+        auto ret = std::move(_full_blocks.front());
+        _full_blocks.pop_front();
+        _full_blocks_size -= ret->size();
+        _full_blocks_blob_cnt -= ret->data().size();
+        return ret;
+    }
+    return std::move(_blk);
+}
+
+std::vector<mutation_ptr> log_appender::mutations() const
+{
+    if (!_blk) { // no data appended
+        return {};
+    }
+    if (_full_blocks.empty()) { // no full block
+        return _blk->mutations();
+    }
+    std::vector<mutation_ptr> ret;
+    for (const auto &block : _full_blocks) {
+        std::copy(block->mutations().begin(), block->mutations().end(), std::back_inserter(ret));
```

Comment:
Why not use `ret.insert(ret.end(), block->mutations().begin(), block->mutations().end())` ?